src/ObjectPool/test/DependencyInjectionExtensionsTest.cs

```diff
@@ -0,0 +1,192 @@
+        using var provider = services.BuildServiceProvider();
+        var optionsMonitor = provider.GetRequiredService<IOptionsMonitor<PoolOptions>>();
+
+        Assert.NotNull(sut);
+        Assert.Equal(64, optionsMonitor.Get(typeof(TestDependency).FullName).Capacity);
+    }
+
+    [Fact]
+    public void AddPool_ServiceAndImplementationType_AddsPool()
+    {
+        var services = new ServiceCollection()
+            .AddSingleton<TestDependency>()
+            .AddPool<ITestClass, TestClass>();
+
+        using var provider = services.BuildServiceProvider();
+        var sut = provider.GetService<ObjectPool<ITestClass>>();
+        var optionsMonitor = provider.GetRequiredService<IOptionsMonitor<PoolOptions>>();
+
+        Assert.NotNull(sut);
+        Assert.Equal(TestDependency.Message, sut!.Get().ReadMessage());
+        Assert.Equal(1024, optionsMonitor.Get(typeof(ITestClass).FullName).Capacity);
+    }
+
+    [Fact]
+    public void AddPool_ServiceAndImplementationTypeWithCapacity_AddsPoolAndSetsCapacity()
+    {
+        var services = new ServiceCollection()
+            .AddSingleton<TestDependency>()
+            .AddPool<ITestClass, TestClass>(options => options.Capacity = 64);
+
+        using var provider = services.BuildServiceProvider();
+        var sut = provider.GetService<ObjectPool<ITestClass>>();
+        var optionsMonitor = provider.GetRequiredService<IOptionsMonitor<PoolOptions>>();
+
+        Assert.NotNull(sut);
+        Assert.Equal(TestDependency.Message, sut!.Get().ReadMessage());
+        Assert.Equal(64, optionsMonitor.Get(typeof(ITestClass).FullName).Capacity);
+    }
+
+    [Fact]
+    public void AddPool_ReturnedPooled_CallsTryReset()
+    {
+        var services = new ServiceCollection()
+            .AddSingleton<TestDependency>()
+            .AddPool<ITestClass, TestClass>();
+
+        using var provider = services.BuildServiceProvider();
+        var sut = provider.GetRequiredService<ObjectPool<ITestClass>>();
+
+        var pooled = sut.Get();
+        sut.Return(pooled);
+
+        Assert.Equal(1, pooled.ResetCalled);
+    }
+
+    [Fact]
+    public void ResolvePooledInstanceDirectly()
+    {
+        var services = new ServiceCollection()
+            .AddSingleton<TestDependency>()
+            .AddPool<ITestClass, TestClass>()
+            .AddScoped(provider => provider.GetRequiredService<ObjectPool<ITestClass>>().Get())
+            ;
+
+        using var provider = services.BuildServiceProvider();
+
+        ITestClass resolved;
+
+        using (var scope = provider.CreateScope())
+        {
+            resolved = scope.ServiceProvider.GetRequiredService<ITestClass>();
+        }
+
+        Assert.NotNull(resolved);
+        Assert.Equal(1, resolved.DisposedCalled);
+    }
+
+    [Fact]
+    public void PooledHelperReturnsScopedInstances_SameScope()
+    {
+        var services = new ServiceCollection()
```

Comment:
I put this in a test but this could also be an extension method directly usable. The usage pattern is nice IMO.